pkg/trace/roundtripper_test.go

```diff
@@ -141,124 +141,6 @@ func TestRoundtripper(t *testing.T) {
 				}
 			},
 		},
-		{
-			name:                  "HoneycombServerOtelClient",
-			initRoundTripperState: honeycombInitRoundTripperState,
-			callRoundTripper:      otelCallRoundTripper,
-			expectedGen: func(traceID, rootID, middleID differs.CustomComparer) []map[string]interface{} {
-				return []map[string]interface{}{
-					{
-						"attributes.app.name":         "gobox",
-						"attributes.service_name":     "gobox",
-						"attributes.app.version":      "testing",
-						"attributes.http.flavor":      "1.1",
-						"attributes.http.method":      "GET",
-						"attributes.http.status_code": int64(200),
-						"attributes.http.url":         differs.AnyString(),
-						"attributes.net.peer.name":    "127.0.0.1",
-						"attributes.net.peer.port":    differs.AnyInt64(),
-						"endTime":                     differs.AnyString(),
-						"name":                        "HTTP GET",
-						"parent.remote":               false,
-						"parent.spanID":               middleID,
-						"parent.traceFlags":           "01",
-						"parent.traceID":              traceID,
-						"spanContext.spanID":          differs.AnyString(),
-						"spanContext.traceFlags":      "01",
-						"spanContext.traceID":         traceID,
-						"spanKind":                    "client",
-						"startTime":                   differs.AnyString(),
-						"SampleRate":                  int64(1),
-					},
-					{
-						"attributes.app.name":     "gobox",
-						"attributes.service_name": "gobox",
-						"attributes.app.version":  "testing",
-						"attributes.trace":        "inner",
-						"endTime":                 differs.AnyString(),
-						"name":                    "inner",
-						"parent.remote":           false,
-						"parent.spanID":           rootID,
-						"parent.traceFlags":       "01",
-						"parent.traceID":          traceID,
-						"spanContext.spanID":      middleID,
-						"spanContext.traceFlags":  "01",
-						"spanContext.traceID":     traceID,
-						"spanKind":                "internal",
-						"startTime":               differs.AnyString(),
-						"SampleRate":              int64(1),
-					},
-					{
-						"attributes.app.name":     "gobox",
-						"attributes.service_name": "gobox",
-						"attributes.app.version":  "testing",
-						"endTime":                 differs.AnyString(),
-						"name":                    "trace-rt",
-						"parent.remote":           false,
-						"parent.spanID":           "0000000000000000",
-						"parent.traceFlags":       "00",
-						"parent.traceID":          "00000000000000000000000000000000",
-						"spanContext.spanID":      rootID,
-						"spanContext.traceFlags":  "01",
-						"spanContext.traceID":     traceID,
-						"spanKind":                "internal",
-						"startTime":               differs.AnyString(),
-						"SampleRate":              int64(1),
-					},
-				}
-			},
-		},
-		{
-			name:                  "OtelServerHoneycombClient",
```

Comment:
I'd started this PR with just the first commit.  It was almost ready to go, but then this test failed.

I could have just deleted this test.  It's not like we care about clients using Honeycomb-style headers these days, right?  But when I dug a bit more I decided it might be best to rip the band-aid off and drop support for Honecyomb-style header propagation entirely.